contracts/compounder/AuraBalVault.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.11;
+
+import { GenericUnionVault } from "./GenericVault.sol";
+
+interface IAuraBalStrategy {
+    function harvest(address _caller, uint256 _minAmountOut) external returns (uint256 harvested);
+}
+
+/**
+ * @title   AuraBalVault
+ * @author  lama.airforce
+ */
+contract AuraBalVault is GenericUnionVault {
+    bool public isHarvestPermissioned = true;
+    mapping(address => bool) public authorizedHarvesters;
+
+    constructor(address _token) GenericUnionVault(_token) {}
+
+    /// @notice Sets whether only whitelisted addresses can harvest
+    /// @param _status Whether or not harvests are permissioned
+    function setHarvestPermissions(bool _status) external onlyOwner {
+        isHarvestPermissioned = _status;
```

Comment:
Should Emit any event?

Reply:
I think we prio keeping diff minimal and FE already have all the events they need.